src/Features/Core/Portable/ConvertForToForEach/AbstractConvertForToForEachCodeRefactoringProvider.cs

```diff
@@ -194,6 +194,23 @@ bool IterationVariableIsUsedForMoreThanCollectionIndex(SyntaxNode current)
                             return true;
                         }
 
+                        // Avoid situations where list[i] is assigned:
+                        // list[i] = 1;
+                        // (list[i], a) = (1, 2);
+                        // list[i] += 1;
+                        // list[i]++;
+                        var c = current.Parent.Parent.Parent;
+                        while (!syntaxFacts.IsStatement(c))
+                        {
+                            if (syntaxFacts.IsLeftSideOfAnyAssignment(c) ||
+                                syntaxFacts.IsOperandOfIncrementOrDecrementExpression(c))
+                            {
+                                return true;
+                            }
```

Comment:
are you basically asking "is this written to"?  if so, we have extensions for that.

Reply:
Thank you! I think it could be same or similar. Let me find and compare.

Reply:
@cyrusnajmabadi, I have reviewed IsWrittenTo.
1. There are two separate implementations: for VB and for C#. They use different signatures: VB involves and semantic models and cancellation tokens.
2. IsWrittenTo considers much more scenarios: variables is a ref parameter, is a foreach variables, etc. I'm not sure we are interested in those scenarios in this case.

So, we definitely can arrange an IsWrittenTo method with a common signature for both languages. Then, we can use this method in this scenario. However, it looks like an extra complication for me. Please let me know you thoughts.

---
In reply to: [294988535](https://github.com/dotnet/roslyn/pull/36542#discussion_r294988535) [](ancestors = 294988535)

Reply:
> So, we definitely can arrange an IsWrittenTo method with a common signature for both languages.

You can use: ISemanticFactsService.IsWrittenTo which is common to both.

Reply:
> IsWrittenTo considers much more scenarios: variables is a ref parameter, is a foreach variables, etc. I'm not sure we are interested in those scenarios in this case.

I think it best conveys what we're really trying to ask.